src/telemetry/metrics/labels.rs

```diff
@@ -203,17 +196,9 @@ impl FmtLabels for Classification {
     }
 }
 
-// ===== impl Direction =====
-
-impl Direction {
-    pub fn new(ctx: ctx::Proxy) -> Self {
-        Direction(ctx)
-    }
-}
-
-impl FmtLabels for Direction {
+impl FmtLabels for ctx::Proxy {
```

Comment:
Hm, this is actually a case where I kind of feel like the newtype provides value, and prevents future bugs if `ctx::Proxy` is used elsewhere, maybe a different set of labels.

Reply:
+1 to @seanmonstar's comment. I'm not sure if I find this change to be clearer than the previous code.

Reply:
`ctx::Proxy` is basically only used for labels and for tap.  I'd either prefer to have a newtype that's local to the transport module and one that's local to the http module, however this makes it easy to change the label for one and not the other, which is undesirable.

What I'm primarily trying to do is avoid having any labeling-specific types used anywhere outside of label formatting.  It's complicated and confusing to have a bunch of redundant types. I'd prefer to remove the Direction type and only introduce a wrapper type if we actually have a need to have redundant representations. I'm betting against ever needing it, though.